src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> resultMap;
+        try {
+            File file = new File(fromFileName);
+            List<String> lines = Files.readAllLines(file.toPath());
+            resultMap = getResultMap(lines);
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file", e);
+        }
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            List<String> resultList = new ArrayList<>();
+            for (String key : resultMap.keySet()) {
+                String stringBuilder = key
+                        + ","
+                        + resultMap.get(key).toString();
+                resultList.add(stringBuilder);
+            }
+            resultList.sort(Collections.reverseOrder());
+            for (String line : resultList) {
+                writer.write(line);
+                writer.newLine();
+            }
+            int result = resultMap.get("supply") - resultMap.get("buy");
```

Comment:
There is a potential for `NullPointerException` here if the keys 'supply' or 'buy' are not present in the `resultMap`. Consider checking for the existence of these keys before performing the subtraction.